### DISCUSSION TITLE
Setting up persistent stores for vscode extension and language libraries.

### DISCUSSION THREAD

**Discussion by timuckun** [devcontainers/community#159] [General]
I started using devcontainers for a python experiment. I am using json file created by vs code. I have two issues. One is that the vs code extensions get and the pip libraries get downloaded every time there is a build. It would speed up things immensely if these could be persisted between builds. I tried the "mount" option to mount a local directory to /usr/lib/......./site-packages but that gets mapped after the container is built so any packages installed by the dockerfile are erased.

In addition I would like a persistent home directory so I can have some of my own dotfiles and scripts available to me and even set my own PATH. 

What's the best way to do this? Do I have to set up my own docker compose or something?

**Comment by bamurtaugh** [comment]
Hi there! Thanks for opening this discussion.

For the first part of your question - I think prebuilds could help: https://containers.dev/guide/prebuild.

Regarding a persistent home directory - @samruddhikhandale I'd love to get your thoughts here!
